Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -62,33 +62,29 @@ WarpX::AddSpaceChargeField (WarpXParticleContainer& pc)
                                      "Error: RZ electrostatic only implemented for a single mode");
 #endif
 
-    // Allocate fields for charge and potential
+    // Allocate the potential phi
     const int num_levels = max_level + 1;
-    Vector<std::unique_ptr<MultiFab> > rho(num_levels);
     Vector<std::unique_ptr<MultiFab> > phi(num_levels);
-    // Use number of guard cells used for local deposition of rho
-    const amrex::IntVect ng = guard_cells.ng_depos_rho;
     for (int lev = 0; lev <= max_level; lev++) {
         BoxArray nba = boxArray(lev);
         nba.surroundingNodes();
-        rho[lev] = std::make_unique<MultiFab>(nba, dmap[lev], 1, ng);
         phi[lev] = std::make_unique<MultiFab>(nba, dmap[lev], 1, 1);
         phi[lev]->setVal(0.);
     }
 
     // Deposit particle charge density (source of Poisson solver)
-    bool const local = false;
     bool const reset = true;
     bool const do_rz_volume_scaling = true;
-    pc.DepositCharge(rho, local, reset, do_rz_volume_scaling);
+    pc.DepositCharge(rho_fp, reset, do_rz_volume_scaling);
+    SyncRho(); // Apply filter, perform MPI exchange, interpolate across levels
```

Comment:
Note that this line automatically enables filtering for this relativistic electrostatic solver. (Filtering was ignored previously.)